bytecomp/translcore.ml

```diff
@@ -1160,15 +1167,21 @@ and transl_apply ?(should_be_tailcall=false) ?(inlined = Default_inline)
         let body =
           match build_apply handle ((Lvar id_arg, optional)::args') l with
             Lfunction{kind = Curried; params = ids; body = lam; attr; loc} ->
```

Comment:
The workings of this whole function are rather obscure.  Are some of the `Pgenval` annotations here over-specific?  For example maybe it's the case that `return` should be bound on line 1169 and then used on line 1172?  (Even if it turns out to always be `Pgenval` this may be clearer.)  I think it might be worth adding some comments.